src/main/scala/mesosphere/marathon/state/RunSpec.scala

```diff
@@ -24,6 +24,22 @@ case class BackoffStrategy(
     maxLaunchDelay: FiniteDuration = 5.minutes,
     factor: Double = 1.15)
 
+/**
+  * Configures role enforcement on runSpecs, used for validation.
+  *
+  * If enforce role is false, the role must be set, but no further checking is done.
+  *
+  * @param enforceRole If true, the role on the runSpec must match the specified role, additionally the
+  *                    acceptedResourceRole field can only contain one of ['*', &lt;role&gt;]
+  * @param validRoles List of valid roles
+  */
+case class RoleEnforcement(
+    enforceRole: Boolean = false,
+    validRoles: Set[String] = Set.empty) {
+
+  def defaultRole = validRoles.head
+}
```

Comment:
Hm, this does not feel right. Also, a set is not sorted so `validRoles.head` seems to invoke undefined behavior for `defaultRole`.

Reply:
Yeah, it doesn't. Maybe it'd be better to have an explicit defaultRole here.

Reply:
Replaced it with an explicit defaultRole